micro-rdk/src/common/data_collector.rs

```diff
@@ -0,0 +1,313 @@
+#![allow(dead_code)]
```

Comment:
Is there actual dead code in this file? If so can this be scoped more narrowly (genuinely asking, as I don't know).

Reply:
This was a mistake. It was only a problem if one moved "data" into the default features and ran `make clippy-esp32`. I'll remove